messaginginapp/src/main/java/io/customer/messaginginapp/gist/data/sse/SseConnectionManager.kt

```diff
@@ -200,4 +308,47 @@ internal class SseConnectionManager(
             connectionState = newState
         }
     }
+
+    private fun subscribeToFlows() {
+        if (timeoutJob == null) {
```

Comment:
`timeoutJob/retryJob` are created once and never recreated. If either collector ever throws (anything other than cancellation), the coroutine completes but the job field still points at the finished job, so subscribeToFlows() won’t spin it back up. For retryJob we at least log the failure, but for timeoutJob we don’t even catch it, so any unexpected exception permanently kills heartbeat monitoring until `stopConnection()` resets the field. 

Could we either (a) wrap both collectors in try/catch that clears the job reference so the next `subscribeToFlows()` call restarts them, or (b) still launch them defensively on `cancellation/exception`?